hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/FsVolumeSpi.java

```diff
@@ -77,6 +77,9 @@
   /** Returns true if the volume is NOT backed by persistent storage. */
   boolean isTransientStorage();
```

Comment:
is this will not valid anymore.?

Reply:
The "isTransientStorage" method is still available. 
In the original code, “isTransient“ and “isTransientStorage” methods are used to determine whether to suport FsDatasetCache, Persistent, Quota, and Movable. FsDatasetCache will be used When the storage type is persistent. NVDIMM is RAM to some extent, which is fast. However, NVDIMM is a persistent storage type.  Then, "isTransient" and "isTransientStorage" used to determine whether to support FsDatasetCache can't meet the requirements. Therefore, we add "isRAM" and "isRAMStorage" methods to decide whether cache is supported or not. And the other functions, such as, Persistent,  Quota, and Movable judged by " isTransient" and "isTransientStorage" methods.

Reply:
Ok. Got it.

Reply:
So, NVDIMM is peristent storage and RAM.

Reply:
> So, NVDIMM is peristent storage and RAM.

yes, that’s right.